compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -3545,28 +3545,23 @@ public BLangNode transform(ModuleXMLNamespaceDeclarationNode xmlnsDeclNode) {
 
     @Override
     public BLangNode transform(ClientDeclarationNode clientDeclarationNode) {
-        BLangClientDeclaration clientDeclaration = (BLangClientDeclaration) TreeBuilder.createClientDeclarationNode();
-
-        clientDeclaration.uri = createSimpleLiteral(clientDeclarationNode.clientUri());
-        clientDeclaration.prefix = createIdentifier(clientDeclarationNode.clientPrefix());
-        clientDeclaration.pos = getPosition(clientDeclarationNode);
-        clientDeclaration.annAttachments = applyAll(clientDeclarationNode.annotations());
-
         BLangClientDeclarationStatement clientDeclarationStatement =
                 (BLangClientDeclarationStatement) TreeBuilder.createClientDeclarationStatementNode();
-        clientDeclarationStatement.clientDeclaration = clientDeclaration;
-        clientDeclarationStatement.pos = getPosition(clientDeclarationNode);
+        Location position = getPosition(clientDeclarationNode);
```

Comment:
Can we directly assign to `clientDeclarationStatement.pos` here and use it instead of `position` since assigning to `clientDeclarationStatement.clientDeclaration` shouldn't affect `clientDeclarationStatement.pos`?

Reply:
But it is the same pos? With a separate variable we avoid accessing the position via `clientDeclarationStatement` for the client declaration or calling getPosition twice.

Reply:
Is accessing via `clientDeclarationStatement.pos` less efficient than using a variable? 
I was wondering if using a separate variable would wrongly imply that position might be altered before being assigned to `clientDeclarationStatement.pos`

Reply:
> Is accessing via `clientDeclarationStatement.pos` less efficient than using a variable?

In this particular case, I don't think it makes much of a difference. 

> I was wondering if using a separate variable would wrongly imply that position might be altered before being assigned to `clientDeclarationStatement.pos`

I don't think so. At least in this code base, this is a practice we've followed. And since it's only used in two places, IMO it's clear what the usage is.